Build failure: furnace

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

```bash
nix run nixpkgs#furnace
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/294735653

### Relevant log output

```console
[ 92%] Building CXX object CMakeFiles/furnace.dir/extern/imgui_patched/backends/imgui_impl_opengl2.cpp.o
/build/source/extern/imgui_patched/backends/imgui_impl_opengl2.cpp:64:10: fatal error: GL/gl.h: No such file or directory
   64 | #include <GL/gl.h>
      |          ^~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/furnace.dir/build.make:6334: CMakeFiles/furnace.dir/extern/imgui_patched/backends/imgui_impl_opengl2.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/furnace.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

The build for this version was working before, not sure what happened:

```bash
✖ (Failed)     furnace-0.6.8.1  2025-04-19  https://hydra.nixos.org/build/294735653
✔              furnace-0.6.8.1  2025-04-12  https://hydra.nixos.org/build/294694179
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.22, NixOS, 25.05 (Warbler), 25.05.20250409.f675531`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.1`
 - nixpkgs: `/nix/store/zd0x43g1jcav0xy0k6q9pdhy3bvif77n-source`

### Notify maintainers

@OPNA2608 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fu/furnace/package.nix
+++ b/pkgs/by-name/fu/furnace/package.nix
@@ -12,6 +12,7 @@
   freetype,
   libsndfile,
   libX11,
+  libGL,
   rtmidi,
   SDL2,
   zlib,
@@ -63,9 +64,11 @@ stdenv.mkDerivation (finalAttrs: {
       libsndfile
       rtmidi
       SDL2
-      libX11
       zlib
       portaudio
+    ]
+    ++ lib.optionals withGL [
+      libGL
     ]
     ++ lib.optionals withJACK [
       libjack2
@@ -73,6 +76,7 @@ stdenv.mkDerivation (finalAttrs: {
     ++ lib.optionals stdenv.hostPlatform.isLinux [
       # portaudio pkg-config is pulling this in as a link dependency, not set in propagatedBuildInputs
       alsa-lib
+      libX11
     ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin (
       with darwin.apple_sdk.frameworks;

